File: ProjectFeedget/app/src/main/java/kr/mashup/projectnoname/ui/main/Contract.java
package kr.mashup.projectnoname.ui.main;

import android.content.Context;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.view.ViewPager;
import android.view.View;

interface Contract {

    interface View {
        Context getContext();
        FragmentManager getSupportFragmentManager();
    }

    interface Presenter {
        void initTabPager(ViewPager pager, TabLayout tabLayout);
        void setWriteClickEvent(android.view.View view);
    }

}

File: ProjectNoname/app/src/main/java/kr/mashup/projectnoname/ui/base/Contract.kt
package kr.mashup.projectnoname.ui.base

import android.app.Activity

interface RootView{

}
interface RootPresenter{

}

interface View<out T : RootPresenter> : RootView{
    val presenter : T
}

interface Presenter<out T : RootView>: RootPresenter{
    val view : T
}
